manager/orchestrator/updater.go

```diff
@@ -111,9 +117,17 @@ func (u *Updater) Run(ctx context.Context, cluster *api.Cluster, service *api.Se
 	}
 	// Abort immediately if all tasks are clean.
 	if len(dirtyTasks) == 0 {
+		if service.UpdateStatus != nil && service.UpdateStatus.State == api.UpdateStatus_RUNNING {
+			u.completeUpdate(ctx, service.ID)
+		}
 		return
 	}
 
+	// If there's no update in progress, we are starting one.
+	if service.UpdateStatus == nil || service.UpdateStatus.State == api.UpdateStatus_NEW {
+		u.startUpdate(ctx, service.ID)
```

Comment:
It's not ideal but it doesn't seem to cause any problems.

Any ideas for a better way?

Seems like it's a tradeoff between making this more graceful/efficient and keeping the PR simple.